runtime/vm/jvminit.c

```diff
@@ -8193,17 +8189,30 @@ freeClassNativeMemory(J9HookInterface** hook, UDATA eventNum, void* eventData, v
 #endif /* defined(J9VM_OPT_OPENJDK_METHODHANDLE) */
 }
 
-#if JAVA_SPEC_VERSION >= 22
+
 static void
 vmHookAnonClassesUnload(J9HookInterface** hook, UDATA eventNum, void* eventData, void* userData)
 {
 	J9VMAnonymousClassesUnloadEvent * unloadedEvent = (J9VMAnonymousClassesUnloadEvent *)eventData;
 	J9VMThread * vmThread = unloadedEvent->currentThread;
+	J9JavaVM *vm = vmThread->javaVM;
 	for (J9Class* j9clazz = unloadedEvent->anonymousClassesToUnload; j9clazz; j9clazz = j9clazz->gcLink) {
+		/* AnonClass->classLoader points to the hostclass->classLoader not the anonClassLoader. */
+		if (J9VM_SHOULD_CLEAR_JNIIDS_FOR_ASGCT(vm, j9clazz->classLoader)) {
+			void **jniIDs = j9clazz->jniIDs;
+			if (NULL != jniIDs) {
+				UDATA size = J9VM_SIZEOF_CLASS_JNIID_TABLE(j9clazz->romClass) / sizeof(void*);
```

Comment:
This should probably be the opposite - the macro should be entries, and the bytewise use should multiply by the pointer size in ensureJNIIDTable.